### DISCUSSION TITLE
Quary regarding adding extra feature in graph

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#3704] [Q&A]
I want to add an extra feature in the data loader, which is not a node feature i.e., these features are for individual graphs. The shape of the features is (# graph in a batch,19). How can I do these?

**Comment by rusty1s** [comment]
You can simply add each global feature as an additional attribute to each `data` object.

**Reply by shrimonmuke0202** [reply]
Thanks for your reply.
Can you give me an example?

**Reply by rusty1s** [reply]
Something like that?
```python
data.global_feat = ...  # e.g., shape [1, 19]
```

**Reply by shrimonmuke0202** [reply]
I want this in a batch, shape should be [batch,19]?

**Reply by rusty1s** [reply]
If you add the feature to each `data` object individually, it will be correctly batched to `[batch, 19]`. Otherwise, it is not clear to me how do you want to obtain the batch-wise global feature matrix in the first place.

**Reply by shrimonmuke0202** [reply]
Thanks, I understood that
